src/matrix.jl

```diff
@@ -259,3 +259,29 @@ function sample(A::IntervalMatrix{T}; rng::AbstractRNG=GLOBAL_RNG) where {T}
     end
     return B
 end
+
+"""
+    diam(A::IntervalMatrix{T}) where {T}
+
+Return a matrix whose entries describe the diameters of the intervals.
+
+### Input
+
+- `A` -- interval matrix
+
+### Output
+
+A matrix `B` of the same shape as `A` such that `B[i, j] == diam(A[i, j])` for
+each `i` and `j`.
+"""
+function diam(A::IntervalMatrix{T}) where {T}
```

Comment:
`diam(::IntervalArithmetic.Interval)` already exists. If we use map the entries, with the implementation `diam(A::IntervalMatrix) = map(diam, A)` this has the added advantage that the matrix type of `B` is the same as `A`.

Reply:
With `diam2(A::IntervalMatrix) = map(diam, A)`, this seems to be slower.
```julia
julia> @btime diam2($A)
  65.611 ns (2 allocations: 128 bytes)
2×2 Array{Float64,2}:
 0.394101  0.378089
 2.22688   1.45347 

julia> @btime diam($A)
  43.957 ns (1 allocation: 112 bytes)
2×2 Array{Float64,2}:
 0.394101  0.378089
 2.22688   1.45347 
```
I leave it to you to decide what is better.

Reply:
Ok. The best choice ultimately depends on the context. I think that using `map` is better here, to keep the structure of the matrix.

Reply:
With the `map` version my test fails. So apparently `map` has a slightly different meaning. I am not sure whether the new meaning is actually correct and the test is wrong or the other way around.

Reply:
The thing is that `diam` is not exactly implemented as `sup(x) - inf(x)` in `IntervalArithmetic`, because the package takes care of correct rounding. See [here](https://github.com/JuliaIntervals/IntervalArithmetic.jl/blob/a1a41da007276ad63ba34d3f1b0d354fd4d5e536/src/intervals/arithmetic.jl#L491).